### PR TITLE
BUG:warn on Nan in minimum,maximum for scalars, float16

### PR DESCRIPTION
**Pull request by mattip**
Fixes #11589 

PR #11043 added a check for Nan in the minimum, maximum loops.

Scalars and `float16` take a different code path through maximum, minimum that was not checked. Tests added that failed before and pass after the fix.

### CODE REVIEW

**Review by tylerjereddy — COMMENTED** [review]
Single CI test failure on posix and windows involving `np.maximum` and `float16` is probably related.

**Comment by mattip** [comment]
Removed code that emits a warning on `float16`, if desired that should be a separate PR

**Comment by seberg** [comment]
The float16 warnings still seem to show up in the tests?

File: numpy/core/tests/test_umath.py [hunk 1]
```diff
@@ -1339,6 +1339,12 @@ def test_reduce_warns(self):
                     assert_equal(np.min(r), np.nan)
                 assert_equal(len(sup.log), n)
 
+    def test_minimize_warns(self):
+        # gh 11589
+        with suppress_warnings() as sup:
+            sup.record(RuntimeWarning)
+            np.minimum(np.nan, 1)
+
```
Comment (seberg) [inline numpy/core/tests/test_umath.py]: `assert_warns` should be fine here? But the test does not actually check that the one warning is recorded like this.

**Comment by charris** [comment]
What was the problem with float16?

**Comment by mattip** [comment]
`float16` never emit a `npy_set_floatstatus_invalid` in `np.maximum`, `np.minimum` even if Nan appears, unlike other floats. In the original PR I added `npy_set_floatstatus_invalid` to the HALF_{maximum, minimum} loops, but this goes beyond the original issue and changes current behavior, so I removed it in a subsequent changeset

**Comment by charris** [comment]
@mattip Sounds like float16 should raise a warning just to be consistent, what is the argument against that?

**Comment by seberg** [comment]
I thought the argument was: "lets not go there for a minor release"? then again, it might have been, lets put in the quick bug fix before the release happens at the time.

I definitely agree that we should enable the warnings for float16, but I would say that now that 15.0 is out, it is simplest to just wait for the next release for the float16 part (I assume the float32 is partially a regression here), it is not a huge deal and has been buggy forever.

**Comment by mattip** [comment]
See PR #11660 which extends this for 1.16 (no backport of that PR needed)

**Pull request merged**